lib/ready.js

```diff
@@ -1,8 +1,10 @@
+import {scheduleMacroTask} from './tasks'
+
 export function whenReady(document, callback) {
   const readyState = document.readyState
   if (readyState === 'interactive' || readyState === 'complete') {
-    callback()
+    scheduleMacroTask(document, callback)
   } else {
-    document.addEventListener('DOMContentLoaded', callback)
+    document.addEventListener('DOMContentLoaded', scheduleMacroTask(document, callback))
```

Comment:
shouldn't the listener be a function calling `scheduleMacroTask(document, callback)` instead ?